app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt

```diff
@@ -193,8 +196,14 @@ class ResourcesGridPresenter(
     suspend fun updateSelection(
         selection: Set<ResourceId>
     ) = withContext(Dispatchers.Default) {
-        this@ResourcesGridPresenter.selection = resources
-            .filter { selection.contains(it.meta.id) }
+        selectionWithDuplicates = this@ResourcesGridPresenter.resources
+            .filter {
+                selection.contains(it.meta.id)
+            }
+        this@ResourcesGridPresenter.selection = selectionWithDuplicates.distinctBy {
+            it.meta.id
+        }
+
```

Comment:
I think, we should not have one more list for keeping ids. Instead, we could attempt to create index-like structure in the presenter, e.g. `rootsById: Map<ResourceId, Roots>` and use it for writing.

In fact, we don't need it too, because all necessary roots are already hidden in `index: IndexAggregation` and we can just use storages as usually. So we don't need to write code for this, just make sure that updating tags and scores for a duplicated id writes to all storages containing this id.

And we also should ensure that `resourcesIds: List<ResourceId>` contains no duplicates. We could use `Set<ResourceId>` but we need `List` for working with positions etc. So, again, probably very few code needed to filter duplicates somewhere in initialization code.

What really should be modified, is the details dialog. We need something like `contains` method in `RootIndex` and `listRoots(id: ResourceId): Set<Path>` method in `IndexAggregation` for smooth implementation of roots information in the details dialog.